### ISSUE TITLE
Farms/nodes filtering is not working correctly in some cases

### ISSUE THREAD

**Issue by mohamedamer453** [threefoldtech/www-mastodon#108]
### Description 

Original Issue #75 

There are nodes that appear as offline in the `Select Node ID` and sometimes they don't appear at all even when they are online in the dashboard.

Both nodes that were mentioned in the [issue](https://github.com/threefoldtech/www-mastodon/issues/75#issuecomment-1337689345) are not appearing when i tried to select them.

Node `321` which belongs to farm `77` is online in the dashboard and the same farm has other online nodes as seen below.

![image](https://user-images.githubusercontent.com/101194226/206510245-6dca1ce0-4f5b-49b1-810e-cf97ce22fa31.png)  

However it shows that there are no online nodes in farm `77`

![image](https://user-images.githubusercontent.com/101194226/206510082-b89384d4-8001-4003-858d-f05d7939f4d9.png)

Same issue for node `340` and farm `76`

![image](https://user-images.githubusercontent.com/101194226/206510554-79fd47c9-6d88-461e-93b0-3d16476f6e04.png)

and when i switched the `Public IPV4` filter the nodes appeared in the farm but it appeared as offline.

![image](https://user-images.githubusercontent.com/101194226/206511151-9031ed8a-c279-4ca2-bf03-752a43b9b9d5.png)

![image](https://user-images.githubusercontent.com/101194226/206511563-49a6f52f-1b98-484e-95ae-1d35b9793b4c.png)

**Comment by samtaggart** [comment]
<img width="297" alt="Screen Shot 2022-12-13 at 3 59 25 PM" src="https://user-images.githubusercontent.com/50134479/207324705-c7027bdf-2163-43f5-8776-2b5b034d193c.png">

... Are there no certified nodes available anywhere? cc: @mohamedamer453

**Comment by mohamedamer453** [comment]
Reopening this issue since the problem still exists. the added fix was to Improve the error message, however the issue mentioned above is not related to the message.

The issue can be seen in detail in the video below. firstly i selected farm `76` and it showed that there are no nodes that satisfy my requirements available in this farm. but when i enabled the `Public IPv4` option the nodes appeared but they appeared as offline as seen in node `340` below.

[Get Mastodon (40).webm](https://user-images.githubusercontent.com/101194226/208294484-5a4e8642-c649-4791-b13a-b26c943e6bc1.webm)

![image](https://user-images.githubusercontent.com/101194226/208294557-596e45a6-c9d9-41df-82df-125cae06a0d8.png)


and the main issue is that node `340` on mainnet is online and has enough resources for the deployment, same thing applies for all nodes that belong to this farm.

![image](https://user-images.githubusercontent.com/101194226/208294887-9511b1f8-9705-448c-bd90-d463178d10f8.png)

The same issue appears also in farm `77`.

And it can also be seen for more than one node in https://github.com/threefoldtech/www-mastodon/issues/109#issuecomment-1356770395.

**Comment by Mahmoud-Emad** [comment]
@mohamedamer453 

The issue you mentioned should be fixed now, the nodes should appear based on the farm id
there is another issue still available 
the node is not responding and it came from the grid proxy maybe because the node is down.
#### NOTE: in the dashboard, the request to update the status of the node happens every 2 hours so if the node is offline you won't know 
and in the mastodon provider, we make the request when you select the farm

**Comment by Mahmoud-Emad** [comment]
verified:

1. steps to produce go to [mastodon](https://getmastodon.threefold.io/)
2. insert your credentials (if the first time insert your ssh key also and reload)
3. wait for the ssh key to load
4. go to the advanced section
5. select farm and you'll see only the nodes exists inside the selected farm will appear 


the second part of this issue as I mentioned not solved yet but seems it came from the grid proxy so I'll WAIT on it until it is fixed in the grid proxy.

**Comment by Mahmoud-Emad** [comment]
verified on [mastodon](https://getmastodon.threefold.io/)
only nodes inside selected farms appear in the node selections, if the node is not reachable the user will see an offline status.
